### PR TITLE
[NEW] Change error.socket.default_error identifier; Alert refactoring

### PR DESCRIPTION
**Pull request by soundstorm**
Change error.socket.default_error_ to error.socket.default_error.message/title to match other messages. Now we can use Alert(key:).

Refactored code to use the Alert helper instead of UIAlertController to reduce redundancy.

Added tests for Alert.swift

@RocketChat/ios

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/ChannelInfoViewController.swift [hunk 1]
```diff
@@ -94,7 +94,7 @@ class ChannelInfoViewController: BaseViewController {
 
     func showStarredList() {
         guard let userId = AuthManager.currentUser()?.identifier else {
-            alert(title: localized("error.socket.default_error_title"), message: "error.socket.default_error_message")
```
Comment (soundstorm) [inline Rocket.Chat/Controllers/Chat/ChannelInfoViewController.swift]: I've never seen this alert, but must be interesting 😅

File: Rocket.Chat/Controllers/Auth/AuthViewController.swift [hunk 2]
```diff
@@ -155,14 +155,9 @@ final class AuthViewController: BaseViewController {
...
-                    title: localized("error.socket.default_error_title"),
-                    message: error["message"]?.string ?? localized("error.socket.default_error_message"),
-                    preferredStyle: .alert
-                )
-
-                alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
-                present(alert, animated: true, completion: nil)
+                Alert(
+                    key: "error.socket.default_error"
+                ).present()
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Auth/AuthViewController.swift]: 👍

File: Rocket.Chat/Controllers/Chat/ChannelInfoViewController.swift (hunk 1 above)
Comment (rafaelks) [reply Rocket.Chat/Controllers/Chat/ChannelInfoViewController.swift]: 😂

**Comment by rafaelks** [comment]
Looking great dude! 👊

File: Rocket.ChatTests/Helpers/AlertSpec.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+//
+//  Created by Luca Justin Zimmermann on 26/01/18.
+//  Copyright © 2018 Rocket.Chat. All rights reserved.
+//
+
+import XCTest
+@testable import Rocket_Chat
+
+class AlertSpec: XCTestCase {
+
```
[the remark this answers is not in the document]
Comment (soundstorm) [reply Rocket.ChatTests/Helpers/AlertSpec.swift]: Bla bla bla... Already fixed, nasty dog 😝

[the remark this answers is not in the document]
Comment (rafaelks) [reply Rocket.ChatTests/Helpers/AlertSpec.swift]: @soundstorm You fixed these issues already?

[the remark this answers is not in the document]
Comment (soundstorm) [reply Rocket.ChatTests/Helpers/AlertSpec.swift]: yep, no more empty lines with spaces.

**Pull request merged**